### ISSUE TITLE
Generating unique labels for topics using LLM

### ISSUE THREAD

**Issue by Pratik--Patel** [MaartenGr/BERTopic#1341]
I am using an LLM based approach described [here ](https://maartengr.github.io/BERTopic/getting_started/representation/llm.html#transformers) to generate labels for the topic words. However, we observe that the labels are not unique across topics and sometimes same label is assigned to multiple topics. This is a problem for us because we would like to distinguish different topics preferably by having unique labels without simple qualification with topic_id etc.

Any idea how it can be achieved? Is it possible to have a prompt where we essentially say that `generate a label for these words but the label should not be any of the following: <list of previously generated labels>` so that labels do not repeat?

**Comment by MaartenGr** [comment]
Hmmm, that might actually sound like a bug. Could you share your full code? Also, which version of BERTopic are you using? That gives me a bit more context to work with.

**Comment by Pratik--Patel** [comment]
version is `0.14.1`

**Comment by manuterui** [comment]
Hi both,

Could you share the code to generate the labels using LLMs? I've used the code below, but it hasn't generated any labels for me as per the [BERTopic documentation](https://maartengr.github.io/BERTopic/getting_started/representation/llm.html#transformers).

`prompt = """
I have topic that contains the following documents: \n[DOCUMENTS]
The topic is described by the following keywords: [KEYWORDS]

Based on the above information, can you give a short label of the topic?
"""

from transformers import pipeline
from bertopic.representation import TextGeneration

# Create your representation model
generator = pipeline('text2text-generation', model='google/flan-t5-base')
representation_model = TextGeneration(generator)

topic_model = BERTopic(representation_model=representation_model)
topic_model.get_topic_info()
`

**Comment by MaartenGr** [comment]
@Pratik--Patel Could you share your full code?

@manuterui You did not `.fit` your model based on the code that you shared. You should train your model on data before it can extract topics.

**Comment by manuterui** [comment]
hi Maarten,
any chance you can share the full code with me (and perhaps on your tutorial page)?
i'm not sure how to integrate :(

```
prompt = """
I have topic that contains the following documents: \n[DOCUMENTS]
The topic is described by the following keywords: [KEYWORDS]

Based on the above information, can you give a short label of the topic?
"""

from transformers import pipeline
from bertopic.representation import TextGeneration

# Create your representation model
generator = pipeline('text2text-generation', model='google/flan-t5-base')
representation_model = TextGeneration(generator)

topic_model = BERTopic(representation_model=representation_model)
topic_model.fit(docs)

topic_model.get_topic_info()
```

**Comment by MaartenGr** [comment]
You just change the following:

```python
generator = pipeline('text2text-generation', model='google/flan-t5-base')
```

To something like this, depending on which model you want to use:

```python
generator = pipeline('text2text-generation', model='google/flan-t5-xxl')
```

**Comment by MaartenGr** [comment]
Closing this due to inactivity. Let me know if you want to re-open the issue!
